rust-runtime/aws-smithy-client/src/http_connector.rs

```diff
@@ -55,26 +56,99 @@ impl HttpConnector {
     }
 }
 
-/// HttpSettings for HTTP Connectors
+/// Builder for [`HttpSettings`].
 #[non_exhaustive]
 #[derive(Default, Debug)]
-pub struct HttpSettings {
-    /// Timeout configuration used when making HTTP connections
-    pub http_timeout_config: timeout::Http,
-    /// Timeout configuration used when creating TCP connections
-    pub tcp_timeout_config: timeout::Tcp,
+pub struct HttpSettingsBuilder {
```

Comment:
We should consider renaming this to `ConnectorSettings` because we use the `connector` terminology in several other places and I believe it to be more specific than the current name.